segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogTests.java

```diff
@@ -205,7 +206,7 @@ public void testAutoCloseOnBookieFailure() throws Exception {
                 // WriteFailureException (general unable to write) should be thrown.
                 AssertExtensions.assertThrows(
                         "First write did not fail with the appropriate exception.",
-                        () -> log.append(new ByteArraySegment(getWriteData()), TIMEOUT),
+                        (Supplier<CompletableFuture<LogAddress>>) () -> log.append(new ByteArraySegment(getWriteData()), TIMEOUT),
```

Comment:
This is really bad. This doesn't even involve an exception. What is giving this warning/error?